### PR TITLE
Creating Okta Breaker Pattern

### PR DESCRIPTION
**Pull request by bastosmichael**
## Description of change
Only adding an Okta Breaker service in order to track when it's down or up etc and eventually create an alert process for notifying us when service is down etc.

https://github.com/department-of-veterans-affairs/vets-contrib/issues/432

## Testing done
Code has been tested against existing tests as well as had been tested in rails console with appropriate okta api tokens to ensure responses are accurate.

## Testing planned
Adding some additional rspec tests for the services themselves to ensure they are covered.

## Acceptance Criteria (Definition of Done)
When All Okta faraday requests are using the Breaker service and still passing all necessary tests. Check to make sure proper metrics are being sent to Prometheus

#### Unique to this PR
- [x] Migrate all Okta Faraday requests to a Breaker service
- [ ] Collect the same kind of metrics we do on requests to Okta
- [ ] Number of reqs/s, errors, uptime, as we do for other third-party services. 
- [ ] Should feed into alerts so we're aware of any unexpected issues and can raise them to okta.

#### Applies to all PRs

- [x] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/okta/configuration.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'common/client/configuration/rest'
+
+module Okta
+  class Configuration < Common::Client::Configuration::REST
+    def base_path
+      'https://deptva-eval.okta.com/api/v1/'
```
Comment (patrickvinograd) [inline lib/okta/configuration.rb]: This should be pulled from the Settings object, not hard-coded.

File: lib/okta/profile/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'okta/configuration.rb'
+
+module Okta
+  module Profile
+    class Configuration < Okta::Configuration
+      def base_path
+        Settings.oidc.profile_api_url || 'https://deptva-eval.okta.com/api/v1/users/'
```
Comment (patrickvinograd) [inline lib/okta/profile/configuration.rb]: If `base_path` is the only thing leading you to have separate OktaService and OktaProfileService, then it's fair game to have base_path be `https://deptva-eval.okta.com/api/v1` and then have a `get_profile` method that sets `url` to `users/ + uid` (or do so with path.join or whatever).

Comment (patrickvinograd) [reply lib/okta/profile/configuration.rb]: And I would not default the base_path; I wouldn't want us to inadvertently make requests to the eval org in production. Practically the risk is very low, we wouldn't have the right token, etc, but I don't think we typically do this elsewhere. If it was done to make specs work, then there is a way to factor out the actual URL and other values from VCR cassettes - so you run them the first time with the real value in settings.local.yml, but it puts in a placeholder value into the recorded request/response.

File: lib/okta/configuration.rb (hunk 1 above)
Comment (bastosmichael) [reply lib/okta/configuration.rb]: Understood

File: lib/okta/profile/configuration.rb (hunk 2 above)
Comment (bastosmichael) [reply lib/okta/profile/configuration.rb]: I'll refactor the base_path requirement in the config to not be set or dynamically changeable before I do anything with VCR or start recording responses.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -56,8 +56,8 @@ def with_okta_configured(&block)
     auth_server_metadata_url: 'https://example.com/oauth2/default/.well-known/oauth-authorization-server',
     issuer: 'https://example.com/oauth2/default',
     audience: 'api://default',
-    profile_api_url: 'https://example.com/api/v1/users/',
-    profile_api_token: 'token'
+    base_api_url: 'https://example.com/',
```
Comment (edpaget) [inline spec/rails_helper.rb]: This seems like it should be the same url as in  spec/lib/oidc/key_service_spec.rb

Comment (bastosmichael) [reply spec/rails_helper.rb]: Yes agreed, updating key_service now to accommodate.

**Pull request merged**